daml-script/export/src/main/scala/com/daml/script/export/TreeUtils.scala

```diff
@@ -58,7 +58,7 @@ object TreeUtils {
     Graphs.topoSort(graph) match {
       case Left(cycle) =>
         throw new IllegalArgumentException(s"Encountered cyclic contract dependencies: $cycle")
-      case Right(sorted) => sorted.map(cid => acs.get(cid).get)
+      case Right(sorted) => sorted.collect(Function.unlift(cid => acs.get(cid)))
```

Comment:
We don’t actually expect to ever have `acs.get` return `None` right? No objection to the change just wondering. I sometimes tend to prefer a crash for unexpected cases to silently swallowing them but seems fine here.

Reply:
We do, if an ACS item has a reference to an unknown contract. E.g. the [`"unknown"` test-case](https://github.com/digital-asset/daml/pull/9542/files/65780dc798b7860e0d97c1a37cfc31d6a4ad8f00#diff-1fd7c179b7a83fed4577905906fd689510299a680d11464fc1a6aad1a60ceb3bR88) triggers this.

You're right that currently this doesn't occur with `daml ledger export` as invoked from the command-line because we guard against it with [this check](https://github.com/digital-asset/daml/pull/9542/files#diff-e376d2f8c626004f50a9317dc11a8944d4d9bc9eb00372ea4cd68628d5c5cfd6R131). However, the next step is to address https://github.com/digital-asset/daml/issues/8773 in which case we need to handle unknown contract ids here.